### DISCUSSION TITLE
CH32v003 peripherals (ADC/OPA)

### DISCUSSION THREAD

**Discussion by Cydget** [ch32-rs/ch32-hal#10] [Q&A]
Hello, in the readme it says the blocking ADC is working for CH32v2 and CH32v3. Is there a reason/difference between them and the CH32v003 for why CH32v003 would not work? I haven't tried building it, but thought I would ask here first. 

Also, the https://github.com/ch32-rs/ch32v00x-hal seems to have the ability to turn on the OPA peripheral. Do you know if there is any plans for the OPA peripheral in this hal the near future for CH32v003?

**Comment by andelf** [comment]
ADC should work. 
I'll port the OPA code soon. 
CH32 has some very different OPAs. So I might need to have 2 or 3 types of OPA implemented.
